### PR TITLE
Merge master into 3pc-batch

### PR DESCRIPTION
**Pull request by mzk-vct**
Merge master into 3pc-batch

### CODE REVIEW

File: sovrin_client/test/anon_creds/test_public_repo.py [hunk 1]
```diff
@@ -85,10 +90,11 @@ def submittedPublicRevocationKey(submittedPublicKeys):
 def testSubmitSchema(submittedSchemaDefGvt, schemaDefGvt):
     assert submittedSchemaDefGvt
     assert submittedSchemaDefGvt.seqId
-    # initial schema didn't have seqNo
-    submittedSchemaDefGvt = submittedSchemaDefGvt._replace(seqId=None)
-    assert submittedSchemaDefGvt == schemaDefGvt
-
+    # initial schema has stub seqno - excluding seqno from comparison
+    def withNoSeqId(schema):
+        schema._replace(seqId=None)
```
Comment (lovesh) [inline sovrin_client/test/anon_creds/test_public_repo.py]: You are missing a return before `schema._replace(seqId=None)` and so the next checks is equivalent to `assert None == None`

File: sovrin_client/test/anon_creds/test_public_repo.py [hunk 2]
```diff
@@ -68,8 +71,10 @@ def submittedPublicKeys(submittedSchemaDefGvtID, publicRepo, publicSecretKey,
                         publicSecretRevocationKey, looper):
     pk, sk = publicSecretKey
     pkR, skR = publicSecretRevocationKey
-    return looper.run(
-        publicRepo.submitPublicKeys(id=submittedSchemaDefGvtID, pk=pk, pkR=pkR))
+    return looper.run(publicRepo.submitPublicKeys(id=submittedSchemaDefGvtID,
```
Comment (lovesh) [inline sovrin_client/test/anon_creds/test_public_repo.py]: You might want to rename `submitPublicKeys` to `submitClaimDef`

File: sovrin_client/anon_creds/sovrin_public_repo.py [hunk 3]
```diff
@@ -114,36 +115,45 @@ def __init__(self, client, wallet):
...
         schema = schema._replace(issuerId=self.wallet.defaultId,
                                  seqId=seqNo)
         return schema
 
-    async def submitPublicKeys(self, id: ID, pk: PublicKey,
-                               pkR: RevocationPublicKey = None) -> (
-            PublicKey, RevocationPublicKey):
-        pkData = pk.toStrDict()
-        pkRData = pkR.toStrDict()
+    async def submitPublicKeys(self,
```
Comment (lovesh) [inline sovrin_client/anon_creds/sovrin_public_repo.py]: Rename it to `submitClaimDef`

File: sovrin_client/test/anon_creds/test_public_repo.py [hunk 4]
```diff
@@ -102,15 +108,17 @@ def testSubmitPublicKey(submittedPublicKeys):
 
 def testGetPrimaryPublicKey(submittedSchemaDefGvtID, submittedPublicKey,
```
Comment (lovesh) [inline sovrin_client/test/anon_creds/test_public_repo.py]: Rename appropriately as mentioned in above comments

File: sovrin_client/test/anon_creds/test_public_repo.py [hunk 5]
```diff
@@ -102,15 +108,17 @@ def testSubmitPublicKey(submittedPublicKeys):
...
                             publicRepo, looper):
-    pk = looper.run(publicRepo.getPublicKey(id=submittedSchemaDefGvtID))
+    pk = looper.run(publicRepo.getPublicKey(id=submittedSchemaDefGvtID,
+                                            signatureType='CL'))
     assert pk == submittedPublicKey
 
 
 def testGetRevocationPublicKey(submittedSchemaDefGvtID,
```
Comment (lovesh) [inline sovrin_client/test/anon_creds/test_public_repo.py]: Rename appropriately as mentioned in above comments

File: sovrin_client/test/cli/test_node.py [hunk 6]
```diff
@@ -116,17 +118,25 @@ def checkNodeConnected(nodes):
         for node in nodes:
             name = newNodeData[ALIAS]
             assert name in node.nodeReg
-
+    timeout = waits.expectedClientConnectionTimeout(
```
Comment (lovesh) [inline sovrin_client/test/cli/test_node.py]: Why do we do this
```
timeout = waits.expectedClientConnectionTimeout(
 util.getMaxFailures(len(philCli.nodeReg))
 )
```
calculation repeatedly?, it wont give a different value

File: sovrin_client/test/cli/test_tutorial_manual.py [hunk 7]
```diff
@@ -201,11 +204,11 @@ def executeGstFlow(name, userCLI, userMap, be, connectedToTest, do, fMap,
                                       None)  # Passing None since its not used
 
         faberSchemaId = ID(SchemaKey('Transcript', '1.2', fMap['target']))
-        faberIssuerKey = userCLI.looper.run(
+        faberIssuerPublicKey = userCLI.looper.run(
```
Comment (lovesh) [inline sovrin_client/test/cli/test_tutorial_manual.py]: Again, rename please

File: sovrin_client/test/anon_creds/test_public_repo.py (hunk 2 above)
Comment (ashcherbakov) [inline sovrin_client/test/anon_creds/test_public_repo.py]: It needs to be renamed in anoncreds. Let's do it later.

File: sovrin_client/test/anon_creds/test_public_repo.py (hunk 4 above)
Comment (ashcherbakov) [inline sovrin_client/test/anon_creds/test_public_repo.py]: It needs to be renamed in anoncreds. Let's do it later.

File: sovrin_client/test/anon_creds/test_public_repo.py (hunk 5 above)
Comment (ashcherbakov) [inline sovrin_client/test/anon_creds/test_public_repo.py]: It needs to be renamed in anoncreds. Let's do it later.

File: sovrin_client/anon_creds/sovrin_public_repo.py (hunk 3 above)
Comment (ashcherbakov) [inline sovrin_client/anon_creds/sovrin_public_repo.py]: It needs to be renamed in anoncreds. Let's do it later.

File: sovrin_client/test/cli/test_tutorial_manual.py (hunk 7 above)
Comment (ashcherbakov) [inline sovrin_client/test/cli/test_tutorial_manual.py]: It needs to be renamed in anoncreds. Let's do it later.

File: sovrin_client/test/cli/test_node.py (hunk 6 above)
Comment (ashcherbakov) [inline sovrin_client/test/cli/test_node.py]: We may have different number of nodes in the pool.

**Pull request merged**